tests/Application.Tests.Integration/Folders/Commands/AddFolderTests.cs

```diff
@@ -0,0 +1,251 @@
+        {
+            RoomId = room.Id,
+            Name = new Faker().Commerce.ProductName(),
+            Capacity = 1
+        };
+        
+        var lockerA = await SendAsync(addLockerACommand);
+        var lockerB = await SendAsync(addLockerBCommand);
+        
+        var addFolderCommand = _folderGenerator.Generate();
+        var addFolderCommandForLockerA =  addFolderCommand with 
+        {
+            Name = sameFolderName,
+            LockerId = lockerA.Id
+        };
+        var addFolderCommandForLockerB =  addFolderCommand with 
+        {
+            Name = sameFolderName,
+            LockerId = lockerB.Id
+        };
+        // Act
+        var folderA = await SendAsync(addFolderCommandForLockerA);
+        var folderB = await SendAsync(addFolderCommandForLockerB);
+        
+        // Assert
+        folderA.Locker.Id.Should().NotBe(folderB.Locker.Id);
+        folderA.Name.Should().Be(folderB.Name);
+        
+        folderB.Name.Should().Be(addFolderCommandForLockerB.Name);
+        folderB.Description.Should().Be(addFolderCommandForLockerB.Description);
+        folderB.Capacity.Should().Be(addFolderCommandForLockerB.Capacity);
+        folderB.Locker.Id.Should().Be(lockerB.Id);
+        folderB.Locker.NumberOfFolders.Should().Be(lockerB.NumberOfFolders + 1);
+        folderB.NumberOfDocuments.Should().Be(0);
+        folderB.IsAvailable.Should().BeTrue();
+        // Clean up
+        var folderAEntity = await FindAsync<Folder>(folderA.Id);
+        var folderBEntity = await FindAsync<Folder>(folderB.Id);
+        var lockerAEntity = await FindAsync<Locker>(lockerA.Id);
+        var lockerBEntity = await FindAsync<Locker>(lockerB.Id);
+        var roomEntity = await FindAsync<Room>(room.Id);
+        Remove(folderAEntity);
+        Remove(folderBEntity);
+        Remove(lockerAEntity);
+        Remove(lockerBEntity);
+        Remove(roomEntity);
+    }
+
+    [Fact]
+    public async Task ShouldThrowConflictException_WhenFolderAlreadyExistsInTheSameLocker()
+    {
+        // Arrange
+        var sameFolderName = new Faker().Commerce.ProductName();
+        var addRoomCommand = _roomGenerator.Generate();
+        var room = await SendAsync(addRoomCommand);
+        
+        var addLockerCommand = _lockerGenerator.Generate();
+        addLockerCommand = addLockerCommand with
+        {
+            RoomId = room.Id,
+            Capacity = 2
+        };
+        var locker = await SendAsync(addLockerCommand);
+        
+        var addFolderCommand = _folderGenerator.Generate();
+        var addFolderCommandForLockerA =  addFolderCommand with 
+        {
+            Name = sameFolderName,
+            LockerId = locker.Id
+        };
+        var addFolderCommandForLockerB =  addFolderCommand with 
+        {
+            Name = sameFolderName,
+            LockerId = locker.Id
+        };
+        // Act
+        var folder = await SendAsync(addFolderCommandForLockerA);
```

Comment:
This too should not be in Act

Reply:
fixed at https://github.com/ProFile-org/ProFile-BE/pull/80/commits/5bf85312dc8fdad26790b428b35e1a66ccad0130